### ISSUE TITLE
Create a table that indicates the supported OS for go-xcat install script

### ISSUE THREAD

**Issue by whowutwut** [xcat2/xcat-core#1616]
I'd like to suggest that we create a matrix table where we indicate the platforms that have been verified by the xCAT team using `go-xcat` to install xCAT on diskful nodes.

We can create this table on RTD and link to it from the http://xcat.org/download.html page

**Comment by neo954** [comment]
I think any Linux distro xCAT support officially should be supported by ``go-xcat``.

If it is not the case, please open a bug with all the details, like the architecture, the name of the Linux distro, and the version of xCAT user intend to install.

**Comment by neo954** [comment]
At the original development stage of ``go-xcat``, only RHEL 6.7, 6.8, 7.1, 7.2, SLES 11.3, 11.4, 12, 12.1, Ubuntu 14.04.4, 16.04, Fedora 23 were tested and verified by myself.

**Comment by daniceexi** [comment]
Our automation will switch to use ``go-xcat`` to install the test env, so that means the ``go-xcat`` will be verified for all the OS/ARCH that have been covered by xCAT automation.  In this case, I think we can claim that ``go-xcat`` can support all the xCAT supported OS/ARCH.

**Comment by immarvin** [comment]
so @whowutwut , what's the final decision, is it ok to close this issue?

**Comment by whowutwut** [comment]
@neo954  When running `go-xcat` on RHEL 6.0, the following error occurs: 

```
[root@c910f04x42vm101 tmp]# /tmp/go-xcat install
Operating system:   linux
Architecture:       x86_64
Linux Distribution: rhel
Version:            6.0

Reading repositories ...... failed
Loaded plugins: rhnplugin
This system is not registered with RHN.
RHN support will be disabled.
No such command: updateinfo. Please use /usr/bin/yum --help
```

So we really should have some sort of yum check for level of yum... or something else to prevent this error and let the user know that we do not support running go-xcat at a certain level. 

Or the solution is to not use options that are not supported by older versions of yum  `updateinfo` is not avail in yum on RHEL 6.0, causing this problem.

**Comment by neo954** [comment]
`yum updateinfo` is provided with the plugin yum-security. This plugin extends yum to allow lists and updates to be limited using security relevant criteria.
In order to use it, install yum-security:

    # yum install yum-security

**Comment by neo954** [comment]
The problem mentioned above was fixed in pull request #1864.

**Comment by whowutwut** [comment]
@neo954 , the fix is not working that was provided by #1864 

On a fresh install of RHEL 6 (compute node), we get no message back initially... 
```
[root@c910f02c05p05 ~]# /tmp/go-xcat install 
Operating system:   linux
Architecture:       ppc64
Linux Distribution: rhel
Version:            6.0

Reading repositories ...... failed

```

Then once the repos are refreshed, we do get the message back... so this is OK 

```
[root@c910f02c05p05 tmp]# yum install yum-security
Loaded plugins: rhnplugin
This system is not registered with RHN.
RHN support will be disabled.
local-rhels6-ppc64-Server                                | 3.7 kB     00:00     
local-rhels6-ppc64-Server/primary_db                     | 2.7 MB     00:00     
local-rhels6-ppc64-ppc64                                 | 3.7 kB     00:00     
local-rhels6-ppc64-ppc64/primary_db                      | 2.7 MB     00:00     
Setting up Install Process
Resolving Dependencies
...
...
================================================================================
Installing:
 yum-plugin-security  noarch  1.1.26-11.el6    local-rhels6-ppc64-Server   25 k

Transaction Summary
================================================================================
Install       1 Package(s)
Upgrade       0 Package(s)

Total download size: 25 k
Installed size: 52 k
Is this ok [y/N]: n
Exiting on user Command
Complete!
[root@c910f02c05p05 tmp]#
```

Then running `go-xcat` again: 
```
[root@c910f02c05p05 tmp]# /tmp/go-xcat install 
Operating system:   linux
Architecture:       ppc64
Linux Distribution: rhel
Version:            6.0

Reading repositories ...... failed
go-xcat: Lacking support of `updateinfo' command for `yum'.
go-xcat: Install the `yum-plugin-security' package and rerun.
```

After installing the `yum-plugin-security` RPM, `go-xcat` fails to install 
```
[root@c910f02c05p05 tmp]# rpm -qa | grep yum-plugin
yum-plugin-security-1.1.26-11.el6.noarch
[root@c910f02c05p05 tmp]# /tmp/go-xcat install 
Operating system:   linux
Architecture:       ppc64
Linux Distribution: rhel
Version:            6.0

Reading repositories ...... failed
go-xcat: Lacking support of `updateinfo' command for `yum'.
go-xcat: Install the `yum-plugin-security' package and rerun.
```

**Comment by neo954** [comment]
Yes, @whowutwut, the first problem can be re-created. But I believe it may not be `go-xcat` related. By run `yum clean metadata` on a clean installed xCAT compute node. I got the following error message. `[Errno 2] No such file or directory: '/var/cache/yum/ppc64/6Server'`. Still need to look into it.

    [root@c910f02c03p23 ~]# ./go-xcat check
    Operating system:   linux
    Architecture:       ppc64
    Linux Distribution: rhel
    Version:            6.0

    Reading repositories ...... failed

    [root@c910f02c03p23 ~]# yum clean metadata
    Loaded plugins: rhnplugin


    [Errno 2] No such file or directory: '/var/cache/yum/ppc64/6Server'

**Comment by neo954** [comment]
BTW, it seems the `yum updateinfo` is introduced from RHEL 6.1.

**Comment by whowutwut** [comment]
Moving to next release. As discussed this is fairly low priority, but if we can either do the following: 

1) Create an array in `go-xcat` that is filled in with mininum supported OS/arch and if we detect < that, quit `go-xcat` with a message `This OS is not supported at this time`.  Then consider support later if some customer specifically requests it.   (No support table is needed)
2) Investigate whether the support for updateinfo can be easily implemented on 6.0. 

I would prefer the 1st option because if no one is using `go-xcat` on older OS that does not provide yum functionality, we do not need to spend any extra effort at this time.

**Comment by whowutwut** [comment]
Thanks I think alerting issues or platforms that we do not support in a error message in `go-xcat` is the best way to avoid another table we need to maintain.
